pkg/agent/config.go

```diff
@@ -173,10 +182,14 @@ func (c *Config) validate() error {
 	erc.Whenf(ec, c.Billing.Clients.HTTP != nil && c.Billing.Clients.HTTP.URL == "", emptyTmpl, ".billing.clients.http.url")
 	erc.Whenf(ec, c.DumpState != nil && c.DumpState.Port == 0, zeroTmpl, ".dumpState.port")
 	erc.Whenf(ec, c.DumpState != nil && c.DumpState.TimeoutSeconds == 0, zeroTmpl, ".dumpState.timeoutSeconds")
-	erc.Whenf(ec, c.Metrics.Port == 0, zeroTmpl, ".metrics.port")
-	erc.Whenf(ec, c.Metrics.LoadMetricPrefix == "", emptyTmpl, ".metrics.loadMetricPrefix")
-	erc.Whenf(ec, c.Metrics.RequestTimeoutSeconds == 0, zeroTmpl, ".metrics.requestTimeoutSeconds")
-	erc.Whenf(ec, c.Metrics.SecondsBetweenRequests == 0, zeroTmpl, ".metrics.secondsBetweenRequests")
+	erc.Whenf(ec, c.Metrics.System.Port == 0, zeroTmpl, ".metrics.system.port")
+	ec.Add(c.Metrics.System.Names.Validate())
+	erc.Whenf(ec, c.Metrics.System.RequestTimeoutSeconds == 0, zeroTmpl, ".metrics.system.requestTimeoutSeconds")
+	erc.Whenf(ec, c.Metrics.System.SecondsBetweenRequests == 0, zeroTmpl, ".metrics.system.secondsBetweenRequests")
+	erc.Whenf(ec, c.Metrics.LFC.Port == 0, zeroTmpl, ".metrics.lfc.port")
+	ec.Add(c.Metrics.LFC.Names.Validate())
+	erc.Whenf(ec, c.Metrics.LFC.RequestTimeoutSeconds == 0, zeroTmpl, ".metrics.lfc.requestTimeoutSeconds")
+	erc.Whenf(ec, c.Metrics.LFC.SecondsBetweenRequests == 0, zeroTmpl, ".metrics.lfc.secondsBetweenRequests")
```

Comment:
Yep - implemented in 86cb9f4a277a5357f95920a82d25b074eef3e60a and c1ae8d3b1ed5c2badcf0d3873b939b6be70137a9 (kept separate so that they can be properly squashed later).